_posts/2018-05-25-ecs-calculator.md

```diff
@@ -0,0 +1,108 @@
+  |m3.medium| 1024| 3765|
+  |m3.large| 2048| 7480|
+  |m3.xlarge| 4096| 15040|
+  |m3.2xlarge| 8192| 30160|
+  |----------------------|
+  |c4.large| 2048| 3765|
+  |c4.xlarge| 4096| 7480|
+  |c4.2xlarge| 8192| 15040|
+  |c4.4xlarge| 16384| 30145|
+  |c4.8xlarge| 36864| 60385|
+  |----------------------|
+  |c3.large| 2048| 3765|
+  |c3.xlarge| 4096| 7480|
+  |c3.2xlarge| 8192| 15040|
+  |c3.4xlarge| 16384| 30145|
+  |c3.8xlarge| 32768| 60385|
+  |----------------------|
+  |r4.large| 2048| 15292|
+  |r4.xlarge| 4096| 30664|
+  |r4.2xlarge| 8192| 61408|
+  |r4.4xlarge| 16384| 122881|
+  |r4.8xlarge| 32768| 245857|
+  |r4.16xlarge| 65536| 491809|
+  |----------------------|
+  |r3.large| 2048| 15299|
+  |r3.xlarge| 4096| 30680|
+  |r3.2xlarge| 8192| 61442|
+  |r3.4xlarge| 16384| 122950|
+  |r3.8xlarge| 32768| 245996|
+  |----------------------|
+  |i2.xlarge| 4096| 30680|
+  |i2.2xlarge| 8192| 61442|
+  |i2.4xlarge| 16384| 122950|
+  |i2.8xlarge| 32768| 245996|
+  |----------------------|
+  |i3.large| 2048| 15292|
+  |i3.xlarge| 4096| 30664|
+  |i3.2xlarge| 8192| 61408|
+  |i3.4xlarge| 16384| 122881|
+  |i3.8xlarge| 32768| 245857|
+  |i3.16xlarge| 65536| 491809|
+  |----------------------|
+  |m5.large| 2048| 7690|
+  |m5.xlarge| 4096| 15586|
+  |m5.2xlarge| 8192| 31377|
+  |m5.4xlarge| 16384| 62960|
+  |m5.12xlarge| 49152| 189294|
+  |m5.24xlarge| 98304| 378665|
+  |----------------------|
+  |c5.large| 2048| 3714|
+  |c5.xlarge| 4096| 7634|
+  |c5.2xlarge| 8192| 15473|
+  |c5.4xlarge| 16384| 31152|
+  |c5.9xlarge| 36864| 70351|
+  |c5.18xlarge| 73728| 140780|
+  |----------------------|
+  |d2.xlarge| 4096| 30664|
+  |d2.2xlarge| 8192| 61408|
+  |d2.4xlarge| 16384| 122881|
+  |d2.8xlarge| 36864| 245857|
+
+  </div>
+
+You can also find this information in yaml format [here](/assets/files/2018-05-25-ecs-calculator/ec2-instance-list.yml)
+
+## How the data can help you to optimally scale and fit ECS containers in an EC2 instance?
+
+
+The number of CPU units specifies the minimum amount of CPU reserved for a container. The CPU units are effectively allocated in containers on an EC2 instance. One container can burst above its allocated units if no other containers are taking up resources. **For instance**, if you have two tasks running on a `t2.medium`, each with 1024 CPU units budgeted for its respective container, one task could take up the whole 2048 CPU units given that the other task is inactive.
+
+Besides, containers also share their unallocated CPU units with other containers on the instance with the same ratio as their allocated amount. **For example**, if you have two tasks running on a `t2.medium` each with 0 CPU units, each container of each task will effectively get 1024 CPU units since no other container has reserved them.
+
+In addition to such benefits of using ECS containers, the data given in the table above can give input to your choices of EC2 instance types or designing ECS containers on your instances.
```

Comment:
This sentence doesn't make too much sense. There is something wrong in the chosen words, especially in the second half. What do you mean by designing (ECS) containers? Definitely you should not be talking about ~~ECS containers~~, rather about **containers on ECS**.